### ISSUE TITLE
parallelConsumer.close() is timing out in 0.5.2.6 and 0.5.2.7 for consumer with set producer, after rebalace 

### ISSUE THREAD

**Issue by BartoszSta** [confluentinc/parallel-consumer#648]
Using "core" parallel consumers, by mistake in one of them when building we are setting producer (which is not used by us).
After updating to from 0.5.2.5 to 0.5.2.7 (but also to 0.5.2.6) we see timeout on following scenario when parallelConsumer.close() is executed:
1. 1st instance of application is starting - parallel consumer is started (setting ordering to unordered, concurrency to 10, consumer connecting to localhost with string deserializers with other defaults, **with producer**, commit interval 5s, commit mode periodic consumer sync) for topic with 2 partitions  - both are assigned.
2. 2nd instance of application is started - parallel consumer is started - gets partition no 1, 1st instance parallel consumer gets instance 1 (using default range assignor)
3. 1st application is being closed, parallelConsumer.close() but fails to close in 20 seconds and throws exception

This happens only in 0.5.2.6 and 0.5.2.7 and only if producer is set in parallel consumer options when building parallel consumer.
If producer is not set closing works.

**Comment by rkolesnev** [comment]
Hi,

I have recreated the bug and identified the root cause - i will try to work out a fix and do a PR in the coming week or so.
It is isolated only to setup when non-transactional Producer is set (even if it is not used).

**Comment by rkolesnev** [comment]
Looks like its a duplicate of #637 - closing it as such.
